src/rich-text-editor.component.js

```diff
@@ -27,6 +27,21 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
     }
   }
 
+  function interceptPaste (event) {
+    // https://developer.mozilla.org/en-US/docs/Web/Events/paste
+    event.preventDefault()
+    event.stopPropagation()
+    let paste = (window.clipboardData).getData('text');
+    paste = props.pasteFn(paste)
+  }
+
+  useEffect(() => {
+    if (props.pasteFn) {
+      divRef.current.addEventListener('paste', interceptPaste)
+      return divRef.current.removeEventListener('paste', interceptPaste)
+    }
+  }, [])
```

Comment:
This should be `[props.pasteFn]`. We should not think of effects as a "mount/unmount" effect, but as a "whenever the inputs change" effect. See https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects for more details